pkg/reconciler/clusterimagepolicy/clusterimagepolicy.go

```diff
@@ -260,6 +269,32 @@ func (r *Reconciler) inlinePolicies(ctx context.Context, cip *v1alpha1.ClusterIm
 			return err
 		}
 	}
+	if cip.Spec.Policy != nil && cip.Spec.Policy.URL != nil {
+		err := r.inlinePolicyURL(ctx, cip.Spec.Policy)
+		if err != nil {
+			logging.FromContext(ctx).Errorf("Failed to read policy url %s: %v", cip.Spec.Policy.URL.String(), err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) inlinePolicyURL(ctx context.Context, policyRef *v1alpha1.Policy) error {
+	logging.FromContext(ctx).Infof("inlining policy url %q", policyRef.URL.String())
+	resp, err := http.Get(policyRef.URL.String())
+	if err != nil {
+		return fmt.Errorf("failed to fetch content from policy url: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return fmt.Errorf("failed to fetch content from policy url with code %q", resp.StatusCode)
+	}
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read policy url response: %w", err)
+	}
+	policyRef.Data = string(data)
```

Comment:
I think we really should add the status field into the CIPs sooner rather than later, this one (not picking on it just reminds me of it again), that if the policy is invalid it will fail at run time. I understand that even when creating inlined CIPs and configmap refs, we need to do a better job validating the policy.

Reply:
Yes, we need to take over the status work again. I can start writing something down for this.